### PR TITLE
chore: replace rocket with axum

### PR DESCRIPTION
**Pull request by bonomat**
this is a preparation for adding the orderbook

### CODE REVIEW

File: coordinator/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ edition = "2021"
 [dependencies]
 anyhow = { version = "1", features = ["backtrace"] }
 atty = "0.2.14"
+axum = "0.6.8"
```
Comment (luckysori) [inline coordinator/Cargo.toml:9]: ❓ Can we remove `rocket` then?

File: coordinator/src/main.rs [hunk 2]
```diff
@@ -68,24 +69,16 @@ async fn main() -> Result<()> {
...
-        )
-        .manage(node)
-        .launch()
-        .await?;
+    let addr = SocketAddr::from((http_address.ip(), http_address.port()));
+    tracing::debug!("listening on http://{}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
```
Comment (luckysori) [inline coordinator/src/main.rs:79]:
```suggestion
        .await?;
```

File: coordinator/src/main.rs [hunk 3]
```diff
@@ -68,24 +69,16 @@ async fn main() -> Result<()> {
...
 
-    tracing::trace!(?mission_success, "Rocket has landed");
+    // tracing::trace!(?mission_success, "Rocket has landed");
```
Comment (luckysori) [inline coordinator/src/main.rs:81]: :wrench: Can we add a similar log message without the rocket analogy?

**Review by luckysori — APPROVED** [review]
Let's give `axum` a go!

**Review by holzeis — APPROVED** [review]
I think you mentioned it on the orderbook PR that you want to replace `rocket` as it does not support websockets. Is that the only reason or are there other advantages of `axium` over `rocket`?

Regarding `websockets`. In itchysats we used `Server Sent Event (SSE)`, because we did not need a bidirectional communication with the client. Do we need that in 10101?

https://blog.logrocket.com/server-sent-events-vs-websockets/

**Comment by bonomat** [comment]
> I think you mentioned it on the orderbook PR that you want to replace `rocket` as it does not support websockets. Is that the only reason or are there other advantages of `axum` over `rocket`?
> 

Yes, websocket is the main reason but also it seems like `axum` has also a better performance as it is just a thin layer on top of `hyper` while `rocket` is a bit _bloated_. 


> Regarding `websockets`. In itchysats we used `Server Sent Event (SSE)`, because we did not need a bidirectional communication with the client. Do we need that in 10101?
> 

SSE caused a lot of pain in the past because of _seemingly_ dropping connections and the missing client support. 
I think `bidirectional` will be useful in the future when we want to support multi-match orders, i.e. a market order which gets executed by multiple times. 

Fun fact: https://github.com/flosse/rust-web-framework-comparison states rocket as outdated framework :D

File: coordinator/Cargo.toml (hunk 1 above)
Comment (bonomat) [reply coordinator/Cargo.toml:9]: definitely

**Pull request merged**